### ISSUE TITLE
All of the losses except the reconstruction loss don't change.

### ISSUE THREAD

**Issue by s-omidikia** [genforce/idinvert#57]
Thank you for your interesting work. 
We used the exact code for our medical dataset without changing any settings. What we have observed is that until iteration 30,000 out of 150,000, all the losses, including adv_loss, d_r_loss, d_f_loss, and d_reg, are fixed, and the only loss being changed is the recon_loss or reconstruction loss. There has not been much change in the generated image so far. 
Is it normal to have these observations up to this iteration or do we need to change the settings? We would be glad if you could provide us with any suggestions. Many thanks in advance.

**Comment by zhujiapeng** [comment]
For your medical dataset, is the generator trained by yourself?

**Comment by s-omidikia** [comment]
> For your medical dataset, is the generator trained by yourself?

Thank you for your consideration
First of all, we trained a conditional StyleGAN from scratch on our medical image dataset and the trained network generates images very well.  
But I must indicate to this point that we only commented this part. 
![photo_2024-01-06_21-10-27](https://github.com/genforce/idinvert/assets/106093823/c9826858-5cb7-4a6a-9347-44a7df2cb3ea)
The reason for commenting these code lines was that as we had to create a new environment in Google Colab and it only provided us with one GPU, using the "gen_nccl_ops.nccl_all_reduce" method, which apparently relates to using multiple GPUs, failed us to execute the code, so we commented that.

**Comment by zhujiapeng** [comment]
Conditional StyleGAN? Do you mean you are using one-hot labels?

**Comment by s-omidikia** [comment]
Yes, one-hot labels are fed to the mapping network, and the point is that the generated image during the GAN-inversion process is a realistic image, but it hasn't converged to exactly reconstruct the original image yet. By the way, does commenting the mentioned code lines affect the process?

**Comment by zhujiapeng** [comment]
1. The initial reconstruction is still a realistic image? After getting the encoded latent codes, did you feed them to the synthesized network directly or fed them to the mapping network? 
2. The commented codes do not affect the process when using one GPU.

**Comment by s-omidikia** [comment]
Thank you for your consideration. 

1. The initial reconstruction wasn't realistic, and it took several iterations to make it more realistic. However, at a certain point in the training process, we observed almost no visually obvious improvements in the generated image anymore. (Of course, we made changes to the code to save and monitor the models and visual results more frequently.)

2. Since we didn't modify the main code, it seems that the encoded latent codes are directly fed into the synthesized network.

Do you think continuing the execution of the code can lead to improvement?
